src/cast_core/src/QuiesceActor.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cast_core/actors/QuiesceActor.hpp>
+
+#include <memory>
+
+#include <yaml-cpp/yaml.h>
+
+#include <bsoncxx/json.hpp>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/database.hpp>
+
+#include <boost/log/trivial.hpp>
+
+#include <gennylib/Cast.hpp>
+#include <gennylib/context.hpp>
+#include <gennylib/quiesce.hpp>
+
+namespace genny::actor {
+
+struct QuiesceActor::PhaseConfig {
+    PhaseConfig(PhaseContext& context) {
+        auto actorDb = context.actor()["Database"].maybe<std::string>();
+        if (!actorDb) {
+            BOOST_THROW_EXCEPTION(
+                InvalidConfigurationException("Must give Database in Actor block."));
+        }
+        dbName = *actorDb;
+    }
+
+    std::string dbName;
+};
+
+
+void QuiesceActor::run() {
+    for (auto&& config : _loop) {
+        for (const auto&& _ : config) {
+            auto quiesceContext = _totalQuiesces.start();
+            BOOST_LOG_TRIVIAL(debug) << "QuiesceActor quiescing cluster.";
+            if (quiesce(_client, config->dbName)) {
```

Comment:
note to circle back here - it seems odd to require a db name to quiesce teh whole cluster

Reply:
This is because my understanding is the collection drop checking is db-specific. At least in the 4.0-and-earlier implementation in the js it was. The new implementation uses `serverStatus` which [says](https://docs.mongodb.com/manual/reference/command/serverStatus/) it provides an overview of the db's state so I assume this added field is also db-specific.

Reply:
An alternative might be to discover all the dbs and then wait for drops on them?